pkg/container/set/set.go

```diff
@@ -218,6 +218,15 @@ func (s Set[T]) Get() (m T, found bool) {
 	return m, length > 0
 }
 
+// AsSlice converts the set to a slice.
+func (s Set[T]) AsSlice() []T {
+	out := make([]T, 0, s.Len())
+	for t := range s.Members() {
+		out = append(out, t)
+	}
+	return out
```

Comment:
This can just be `slices.Collect(s.Members())`:

```suggestion
	return slices.Collect(s.Members())
```

Reply:
Any value in pre-allocating the right length?

Reply:
Hm, that's a fair point. We might avoid some re-allocations which could be beneficial if the method is used in a often-exercised hot path. I see the method is mainly used in tests and log messages (where we could avoid allocating the entire slice and then truncating it, also see XXX). The only other use is in `(*policyImporter).updatePrefixes` for which I can't tell how often it is hit. I'll leave it up to you to keep `slices.Collect` or revert back to the pre-allocation.

Reply:
It's not a particularly hot code path, we can leave it as-is.

Reply:
FWIW, the one-liner to have the best of both worlds is `slices.AppendSeq(make(), it)`. `slices.Collect()` [just uses that with a nil slice](https://cs.opensource.google/go/go/+/refs/tags/go1.23.3:src/slices/iter.go;l=58).